### PR TITLE
Fav links

### PR DESCRIPTION
**Pull request by Gallup93**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
## RSpec Results
```
Finished in 1.95 seconds (files took 4.69 seconds to load)
22 examples, 0 failures

Coverage report generated for RSpec to /Users/michael/turing/3module/projects/local_spotlight/coverage. 317 / 339 LOC (93.51%) covered.
```

### CODE REVIEW

File: app/controllers/favorites_controller.rb [hunk 1]
```diff
@@ -2,4 +2,20 @@ class FavoritesController < ApplicationController
...
+  def new
+    if UserArtist.create(user_id: current_user[:id], artist_id: params["id"])
+      flash[:success] = "#{find_artist_by_id(params["id"]).name} has been added to your favorites"
+      redirect_to request.referrer
+    end
+  end
+
+  def destroy
+    favorite = UserArtist.find(params[:id])
+    name = Artist.find(favorite.artist_id).name
```
Comment (Joshua-Tukman) [inline app/controllers/favorites_controller.rb:15]: Great job on this destroy action. Happy you got this figured out

File: app/controllers/favorites_controller.rb [hunk 2]
```diff
@@ -2,4 +2,20 @@ class FavoritesController < ApplicationController
   def index
     @favorite_artists = current_user.user_artists
   end
+
+  def new
+    if UserArtist.create(user_id: current_user[:id], artist_id: params["id"])
+      flash[:success] = "#{find_artist_by_id(params["id"]).name} has been added to your favorites"
+      redirect_to request.referrer
```
Comment (Joshua-Tukman) [inline app/controllers/favorites_controller.rb:9]: Never knew about this .referrer rails method!

Comment (cgaddis36) [reply app/controllers/favorites_controller.rb:9]: never seen it before either!

File: app/models/user.rb [hunk 3]
```diff
@@ -3,6 +3,11 @@ class User < ApplicationRecord
   has_many :artists, through: :user_artists
 
   def find_closest_artists
+    @artists = Artist.all
     # MapService.new(current_user.zipcode)
   end
+
+  def favorited?(user_id, artist_id)
+    User.find(user_id).artists.any? { |artist| artist.id == artist_id}
+  end
```
Comment (cgaddis36) [inline app/models/user.rb:12]: nice use of some ruby action here, potential refactor to Activerecord

File: spec/features/favorites/favorites_new_spec.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      end
+
+      expect(page).to have_content("#{@artist3.name} has been added to your favorites")
+
+      visit "/favorites"
+
+      expect(page).to have_content(@artist1.name)
+      expect(page).to have_content(@artist3.name)
+    end
+    it "wont display 'FAV' link if artist is already facortied" do
```
Comment (cgaddis36) [inline spec/features/favorites/favorites_new_spec.rb:34]: small spelling error

File: spec/features/favorites/favorites_new_spec.rb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      within ".fav-link" do
+        click_link "FAV"
+      end
+
+      expect(current_path).to eq("/artists")
+      expect(page).to have_content("#{@artist1.name} has been added to your favorites")
+      expect(page).to_not have_link("FAV")
+      click_link "#{@artist3.name}"
+      expect(page).to have_link("FAV")
+    end
```
Comment (cgaddis36) [inline spec/features/favorites/favorites_new_spec.rb:52]: nice job testing each favorites path here

Comment (Gallup93) [reply spec/features/favorites/favorites_new_spec.rb:52]: fixed. good catch

File: spec/features/favorites/favorites_new_spec.rb (hunk 4 above)
Comment (Gallup93) [reply spec/features/favorites/favorites_new_spec.rb:34]: fixed. good catch

**Pull request merged**